packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,366 @@
+          </div>
+        </div>
+      </div>
+      <Editors
+        formData={formData}
+        setFormData={setFormData}
+        schema={schema}
+        setSchema={setSchema}
+        uiSchema={uiSchema}
+        setUiSchema={setUiSchema}
+        extraErrors={extraErrors}
+        setExtraErrors={setExtraErrors}
+        setShareURL={setShareURL}
+      />
+      <div className='col-sm-5'>
+        <ErrorBoundary>
+          {showForm && (
+            <DemoFrame
+              head={
+                <>
+                  <link rel='stylesheet' id='theme' href={stylesheet || ''} />
+                  {theme === 'antd' && (
+                    <div
+                      dangerouslySetInnerHTML={{
+                        // @ts-ignore
+                        __html: document.getElementById('antd-styles-iframe')?.contentDocument.head.innerHTML,
+                      }}
+                    />
+                  )}
+                </>
+              }
+              style={{
+                width: '100%',
+                height: 1000,
+                border: 0,
+              }}
+              theme={theme}
+            >
+              <FormComponent
+                {...templateProps}
+                {...liveSettings}
+                schema={schema}
+                uiSchema={uiSchema}
+                formData={formData}
+                onChange={onFormDataChange}
+                noHtml5Validate={true}
+                onSubmit={({ formData }: { formData: string }, e: any) => {
+                  console.log('submitted formData', formData);
+                  console.log('submit event', e);
+                  window.alert('Form submitted');
+                }}
+                fields={{ geo: GeoPosition }}
+                validator={validators[validator]}
+                onBlur={(id: string, value: string) => console.log(`Touched ${id} with value ${value}`)}
+                onFocus={(id: string, value: string) => console.log(`Focused ${id} with value ${value}`)}
+                onError={log('errors')}
+                ref={playGroundFormRef}
+              />
+            </DemoFrame>
+          )}
+        </ErrorBoundary>
+      </div>
+    </>
+  );
+};
+
+export default Playground;
+
+type EditorsProps = {
+  schema: unknown;
+  setSchema: React.Dispatch<React.SetStateAction<object>>;
+  uiSchema: unknown;
+  setUiSchema: React.Dispatch<React.SetStateAction<object>>;
+  formData: unknown;
+  setFormData: React.Dispatch<React.SetStateAction<object>>;
+  extraErrors: unknown;
+  setExtraErrors: React.Dispatch<React.SetStateAction<unknown>>;
+  setShareURL: React.Dispatch<React.SetStateAction<string | null>>;
+};
+
+const Editors: React.FC<EditorsProps> = ({
+  extraErrors,
+  formData,
+  schema,
+  uiSchema,
+  setExtraErrors,
+  setFormData,
+  setSchema,
+  setShareURL,
+  setUiSchema,
+}) => {
+  const onSchemaEdited = useCallback((newSchema) => {
+    setSchema(newSchema);
+    setShareURL(null);
+  }, []);
```

Comment:
Oh oops, you'll want the set functions as dependencies too since they are props
```suggestion
  }, [setSchema, setShareURL]);
```